dashboard/app/models/script.rb

```diff
@@ -1628,4 +1629,51 @@ def all_descendant_levels
     sublevels = levels.map(&:all_descendant_levels).flatten
     levels + sublevels
   end
+
+  def serialize_lessons
+    # include: '**' allows serialization of associations recursively for any number of levels.
+    # https://github.com/rails-api/active_model_serializers/issues/968#issuecomment-557513403s
+    ScriptLessons::ScriptSerializer.new(self).as_json(include: '**')
+  end
+
+  def self.update_lessons_from_seed_files(filenames)
+    lessons_to_import = filenames.map {|filename| Script.lessons_to_import_from_seed_file(filename)}.flatten
+    Lesson.import! lessons_to_import, on_duplicate_key_update: :all
+  end
+
+  LessonFullKey = Struct.new(:key, :lesson_group_key)
+
+  def self.lessons_to_import_from_seed_file(filename)
+    script_lessons_hash = JSON.parse(File.read(filename))
+
+    # We take some care here to minimize the number of SQL queries required.
+    # This method should require 3 SELECTs and 1 INSERT INTO ... ON DUPLICATE KEY UPDATE:
+    #
+    # 1. select the script by name
+    # 2. select that script's lesson groups
+    # 3. select the lessons from those lesson groups
+    # 4. insert with the updated values
```

Comment:
Thank you for this explanation! This will be useful for anyone trying to optimize later.